src/ray/gcs/gcs_server/ray_sync.h

```diff
@@ -0,0 +1,114 @@
+// Copyright 2022 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#pragma once
+
+#include <memory>
+#include <type_traits>
+
+#include "ray/common/asio/periodical_runner.h"
+#include "ray/gcs/gcs_server/gcs_resource_report_poller.h"
+#include "ray/gcs/gcs_server/grpc_based_resource_broadcaster.h"
+
+namespace ray {
+namespace sync {
+
+class RaySync {
+ public:
+  RaySync(instrumented_io_context &main_thread,
+          std::unique_ptr<::ray::gcs::GrpcBasedResourceBroadcaster> braodcaster,
+          std::unique_ptr<::ray::gcs::GcsResourceReportPoller> poller)
+      : ticker_(main_thread),
+        broadcaster_(std::move(braodcaster)),
+        poller_(std::move(poller)) {}
+
+  void Start() {
+    poller_->Start();
```

Comment:
Move the impl to .cc file?

Reply:
Hmm, I think it's ok? Right now we don't have a standard for this I think. Btw, this will be updated in the follow up PR.